### PR TITLE
fix: improved trade button

### PR DESCRIPTION
**Pull request by shafouz**
Rewrote the function that makes the search queries, now:
screenshot-20231220-150009Z-selected

only searches for the conqueror that you chose:

screenshot-20231220-145544Z-selected

but using:

screenshot-20231220-150523Z-selected

has the same behaviour as before:
screenshot-20231220-145601Z-selected

and more importantly, there should not be a problem of the query size getting too big, since we trim if it goes beyond a limit

```javascript
if (final_query.length > max_query_length) {
    final_query = final_query.slice(0, max_query_length)
}
```

### CODE REVIEW

File: frontend/src/lib/skill_tree.ts [hunk 1]
```diff
@@ -384,87 +384,72 @@ const tradeStatNames: { [key: number]: { [key: string]: string } } = {
...
 
-  // POE Trade has a 32k limit
-  if (url.toString().length > 32000) {
-    console.warn('URL too long, generating only selected conqueror');
-    url.searchParams.set('q', JSON.stringify(constructQuery(jewel, conqueror, results, true)));
+  if (final_query.length > max_query_length) {
+    final_query = final_query.slice(0, max_query_length)
   }
 
+  return gen_final_query(final_query)
```
Comment (Vilsol) [inline frontend/src/lib/skill_tree.ts:448]: Is there a reason why this is a function, if it only gets called once? Seems to just add confusion and misdirection.

File: frontend/src/routes/tree/+page.svelte [hunk 2]
```diff
@@ -455,7 +455,7 @@
               {#if results}
                 <button
                   class="p-1 px-3 bg-blue-500/40 rounded disabled:bg-blue-900/40 mr-2"
-                  on:click={() => openTrade(searchJewel, searchConqueror, searchResults.raw)}
+                  on:click={() => openTrade(searchJewel, searchConqueror, searchResults.raw, true)}
```
Comment (Vilsol) [inline frontend/src/routes/tree/+page.svelte:458]: I would still like to retain the logic, that if it can, it should generate for all conquerors.

**Review by Vilsol — CHANGES_REQUESTED** [review]

**Comment by shafouz** [comment]
Changed as requested, the only difference is that now it shows a little less results. I set the query limit by trial and error but it should rarely fail if it does

**Comment by Vilsol** [comment]
It seems like you just removed the retry code and added a filter in a different way?

**Comment by shafouz** [comment]
Yes, instead of checking if the url size is too big and handling it, now we trim it down to a size that should work always

**Comment by Vilsol** [comment]
Yes, but previously, if all conquerors would fit in the query, it would generate one for all conquerors. If not, it falls back to a single one and trims it.

When they fit:
image

When they don't:
image

In your edit, it now generates one giant list containing all of them, but it cuts off.

image

The tool should always prioritize showing the most available seeds for the conqueror selected.

**Comment by shafouz** [comment]
Oh, i see. I'll change to that.

**Comment by shafouz** [comment]
All seeds:
screenshot-20231220-183821Z-selected

Single seed:
screenshot-20231220-183807Z-selected

The maximum amount of seeds in a query is 60 per conqueror before it stops working. If there is more than that should it fall back to the single conqueror case?

**Comment by Vilsol** [comment]
Please make the linter happy and also rename the PR to follow conventional commit guidelines.

**Review by Vilsol — APPROVED** [review]

**Pull request merged**